core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackImageService.java

```diff
@@ -93,7 +93,18 @@ public Image getCurrentImage(Stack stack) throws CloudbreakImageNotFoundExceptio
     public void changeImageCatalog(Stack stack, String imageCatalog) {
         try {
             Image currentImage = componentConfigProviderService.getImage(stack.getId());
+            ImageCatalog sourceImageCatalog = imageCatalogService.getImageCatalogByName(stack.getWorkspace().getId(), currentImage.getImageCatalogName());
+            if (imageCatalogService.isCustomImageCatalog(sourceImageCatalog)) {
+                throw new CloudbreakServiceException(
+                        String.format("Changing custom image catalog '%s' is not supported.", sourceImageCatalog.getName()));
```

Comment:
Misleading error message - a custom catalog is a custom catalog, it's format is irrelevant. This should be something like "migrating from a non-JSON based catalog to a JSON based one is not supported" or something like that.